Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/TapToPlace/TapToPlace.cs

```diff
@@ -0,0 +1,467 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Physics;
+using Microsoft.MixedReality.Toolkit.SpatialAwareness;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using Microsoft.MixedReality.Toolkit.Utilities.Solvers;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Experimental.Utilities
+{
+    /// <summary>
+    /// Tap to place is a far interaction component used to place objects on a surface.
+    /// </summary>
+    public class TapToPlace : Solver, IMixedRealityPointerHandler
+    {
+        [Experimental]
+        [SerializeField]
+        [Tooltip("The game object that will be placed if it is selected")]
+        private GameObject gameObjectToPlace;
+
+        /// <summary>
+        /// The game object that will be placed if it is selected
+        /// </summary>
+        public GameObject GameObjectToPlace
+        {
+            get => gameObjectToPlace;
+            set
+            {
+                if (value != null && gameObjectToPlace != value)
+                {
+                    gameObjectToPlace = value;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Check if a collider is present on the GameObjectToPlace
+        /// </summary>
+        public bool ColliderPresent
+        {
+            get
+            {
+                if (GameObjectToPlace.GetComponent<Collider>() == null)
+                {
+                    return false;
+                }
+
+                return true;
+            }
+        }
```

Comment:
If this is only used in start and the inspector, this shouldn't really be a public interface. 

Mark it as protected and InternalsVisisbleTo for the inspector
https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.internalsvisibletoattribute?view=netframework-4.8

also simplify to 

`protected bool IsColliderPresent => GameObjectToPlace != null ? GameObjectToPlace.GetComponent<Collider>() != null : false;`

Reply:
Also in case you're tempted to simplify to:

```c#
protected bool IsColliderPresent => GameObjectToPlace?.GetComponent<Collider>() != null;
```

Don't do that - Unity's null checking overloading makes that a dangerous thing to do.

Reply:
So yeah @Troy-Ferrell's line is good.

Reply:
`ColliderPresent` is now `protected internal bool IsColliderPresent`